src/state/actions/companionWindow.js

```diff
@@ -72,6 +72,8 @@ export function popOutCompanionWindow(windowId, panelType, position) {
 
     if (position === 'left') {
       dispatch(setWindowSideBarPanel(windowId, panelType));
+
```

Comment:
I know this is still not fully resolved, but I believe the point on using multiple dispatches for state management has reached agreement here https://github.com/ProjectMirador/mirador/issues/2049#issuecomment-470148920

We want making actionCreators that create actions, not cause side effects. Redux is meant to operate on inversion of control; this pattern of multiple dispatches reverses the inversion of control and makes us essentially use function calls to update state, meaning that we need to know all necessary state update logic in the actionCreators.

The [redux docs](https://redux.js.org/recipes/structuring-reducers/beyond-combinereducers#sharing-data-between-slice-reducers) give several patterns for "updating state across slices/branches". The first is to refactor the state. The second pattern, which I think makes the most sense, is to add more data to the action that is created that will allow multiple reducer listeners to have what they need to make the transformation. We can still use `getState()` in action creators to make this data available, but never dispatch more than one action that does not have the same name as the actionCreator. An example of this is given here https://github.com/ProjectMirador/mirador/issues/2049#issuecomment-469951071

The exception to this last rule is for async functions, whose main bodies never dispatch any actions, but only initiate requests that themselves have meaningfully-named actionCreator side-effects. See: https://github.com/ProjectMirador/mirador/pull/2035#issuecomment-469816402

Reply:
We spent a good hour yesterday thinking about how to not do this; it seems to me, until we come to an agreement about the new approach, this is the pattern we've been using and can easily be refactored later.

Reply:
Yeah, totally agree that the multiple dispatch is sub-optimal.  But as @cbeer mentions, at this point, it's all encapsulated in this one place since we're already doing multiple dispatches in this action.

Once the pattern that we'll use for this type of thing emerges, refactoring this piece of code to use that new pattern should be relatively easy (as opposed to us trying to figure that pattern now).  I think we should certainly discourage any new actions being added/modified to include addl. dispatches.

Reply:
@aeschylus you bring up some going points thanks for linking those tickets. Until the larger question is resolved can we move this work forward?

The action creator in question is already dispatching additional actions, adding one more doesn't holistically change the already existing pattern happening. I'm sure the team would be open to addressing these questions at a macro level, in which this work can be refactored, no?

Reply:
Will merge. Thanks for considering guys.